src/libman/Contracts/HostInteraction.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.LibraryManager.Contracts;
+
+namespace Microsoft.Web.LibraryManager.Tools.Contracts
+{
+    /// <inheritdoc />
+    internal class HostInteraction : IHostInteractionInternal
+    {
+        public HostInteraction(EnvironmentSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            Logger = settings.Logger ?? throw new ArgumentNullException(nameof(settings.Logger));
+
+            WorkingDirectory = settings.CurrentWorkingDirectory;
+            CacheDirectory = settings.CacheDirectory;
+        }
+
+        /// <inheritdoc />
+        public string WorkingDirectory { get; private set; }
+
+        /// <inheritdoc />
+        public string CacheDirectory { get; }
+
+        /// <inheritdoc />
+        public ILogger Logger { get; }
+
+        /// <inheritdoc />
+        public async Task<bool> WriteFileAsync(string path, Func<Stream> content, ILibraryInstallationState state, CancellationToken cancellationToken)
```

Comment:
Move HostInteration should implement our existing Interface IHostInteraction
Also operations to read and write should be the same as the other two extensions. Do you want move read/write operations to a common HostIntegationBase implementation of IHostIntegration. This way we ensure all extensions are wrtting/reading consistently